src/app/api/semesters/[id]/route.ts

```diff
@@ -0,0 +1,137 @@
+import type { NextRequest } from "next/server";
+import { NextResponse } from "next/server";
+import { eq } from "drizzle-orm";
+import { z } from "zod";
+
+import { db } from "@/lib/db";
+import { semesters } from "@/lib/db/schema";
+import { getCurrentUser } from "@/lib/session";
+import { updateSemesterSchema } from "@/lib/validators";
+
+export async function GET(
+  _req: NextRequest,
+  { params }: { params: { id: number } },
```

Comment:
```suggestion
  { params }: { params: { semesterId: number } },
```

Reply:
Make the route id [semesterId] instead of [id]? You'll have to rename the id passed in through the params as well